### PR TITLE
Implement an Embedded Pulsar Job Runner

### PR DESCRIPTION
**Pull request by jmchilton**
This implements an initial embedded Pulsar job runner (the last item on #1378 and outlined in [this Trello card](https://trello.com/c/4YwVZBtq/1865-embedded-pulsar-job-runner)). See documentation updates to `job_conf.xml.sample_advanced` for information on configuration.

In addition to the commit that implements this, PR #2052 is included (since this work builds on that) as well smaller doc and import cleanups for the pulsar job runner code in  `lib/galaxy/jobs/runners/pulsar.py`.

### CODE REVIEW

**Comment by natefoo** [comment]
For anyone unclear on why this is cool, this comment from the sample job conf explains it:

> The embedded Pulsar runner starts a Pulsar app internal to Galaxy and communicates it directly. This maybe be useful for instance when Pulsar staging is important but a Pulsar server is unneeded (most obviously for instance if compute servers cannot mount Galaxy's files but Galaxy can mount a scratch directory available on compute).

In other words, you get all of Pulsar's rewriting and staging functionality directly in Galaxy (it can't stage to remote resources, but it can stage all of a job's necessary files to/from a scratch filesystem).

**Comment by bgruening** [comment]
Awesome! :100:

File: config/job_conf.xml.sample_advanced [hunk 1]
```diff
@@ -98,6 +98,26 @@
...
+               internal to Galaxy and communicates it directly.
+               This maybe be useful for instance when Pulsar
+               staging is important but a Pulsar server is
+               unneeded (most obviously for instance if compute
+               servers cannot mount Galaxy's files but Galaxy
+               can mount a scratch directory available on
+               compute). -->
+          <!-- Specify a complete description of the Pulsar app
+               to create. Currently this configuration (if set)
+               must create exactly on job manager. For more
```
Comment (mvdbeek) [inline config/job_conf.xml.sample_advanced:112]: s/on/one ?

File: config/job_conf.xml.sample_advanced [hunk 2]
```diff
@@ -98,6 +98,26 @@
                deprecated and will disappear with a future release of Galaxy.
           -->
         </plugin>
+        <plugin id="pulsar_embedded" type="runner" load="galaxy.jobs.runners.pulsar:PulsarEmbeddedJobRunner">
+          <!-- The embedded Pulsar runner starts a Pulsar app
+               internal to Galaxy and communicates it directly.
+               This maybe be useful for instance when Pulsar
+               staging is important but a Pulsar server is
+               unneeded (most obviously for instance if compute
```
Comment (mvdbeek) [inline config/job_conf.xml.sample_advanced:106]: s/most obviously // ?

File: config/job_conf.xml.sample_advanced (hunk 1 above)
Comment (jmchilton) [inline config/job_conf.xml.sample_advanced:112]: Yup, will open a PR with fix. Thanks.

File: config/job_conf.xml.sample_advanced (hunk 2 above)
Comment (jmchilton) [inline config/job_conf.xml.sample_advanced:106]: Sure - entirely fair point.

**Pull request merged**